### PR TITLE
Implements DNA locks and crushing to RIGs, again

### PR DESCRIPTION
**Pull request by Geevies**
* Adds a DNA lock feature which can be used for different purposes in the future.
* Adds RIG crushing, which applies 10 brute to the wearer every process().
* The ninja's DNA gets attached to its RIG at spawn, if anyone tries to wear it, they'll get crushed, die, and explode. The ninja changing race, colour, hairstyle, etc. does not trigger a DNA change, so the RIG will still see them as the owner. Beware radiation.

I hate merge conflicts.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/clothing/spacesuits/rig/rig.dm [hunk 1]
```diff
@@ -357,13 +365,20 @@
...
 				module.deactivate()
 			offline = 2
 			slowdown = offline_slowdown
 		return
 
+	if(crushing)
+		wearer.apply_damage(10) // Applies 10 brute damage to a random extremity each process
+		if(wearer.stat == DEAD)
+			crushing = FALSE
+			visible_message(span("danger", "A squelching sound comes from within the sealed hardsuit..") // It'll explode at this stage, deleting the suit and the person and the immediate surroundings
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/rig.dm]:
```suggestion
			visible_message(span("danger", "A squelching sound comes from within the sealed hardsuit..")) // It'll explode at this stage, deleting the suit and the person and the immediate surroundings
```

File: code/modules/clothing/spacesuits/rig/rig.dm [hunk 2]
```diff
@@ -301,6 +304,11 @@
 	if (has_sealed_state)
 		icon_state = canremove ? initial(icon_state) : "[initial(icon_state)]_sealed"
 
+	if(dnaLock && !offline)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/spacesuits/rig/rig.dm]: Crush procedures sounds really weird

File: code/modules/clothing/spacesuits/rig/rig.dm [hunk 3]
```diff
@@ -357,13 +365,20 @@
...
+		crushing = FALSE
 		if(offline == 1)
 			for(var/obj/item/rig_module/module in installed_modules)
 				module.deactivate()
 			offline = 2
 			slowdown = offline_slowdown
 		return
 
+	if(crushing)
+		wearer.apply_damage(10) // Applies 10 brute damage to a random extremity each process
```
Comment (NonQueueingMatt) [inline code/modules/clothing/spacesuits/rig/rig.dm]: This means ninja rig explosions, yeah?

File: code/datums/outfits/outfit_antag.dm [hunk 4]
```diff
@@ -137,6 +137,10 @@
 	if(visualsOnly)
 		return
 
+	var/obj/item/weapon/rig/light/ninja/rig = new /obj/item/weapon/rig/light/ninja(src)
```
Comment (Geevies) [inline code/datums/outfits/outfit_antag.dm]:
```suggestion
	var/obj/item/rig/light/ninja/rig = new /obj/item/rig/light/ninja(src)
```

**Comment by alsoandanswer** [comment]
delam the sm to immediately kill the ninja and win round

**Pull request merged**